walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,151 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.*;
+import java.util.stream.Stream; 
+import java.util.stream.Collectors;
 
+/*
+  Finds times for a meeting to take place
+*/
 public final class FindMeetingQuery {
+
+  // The end of the day, i.e. 1440 minutes. 
+  private final int LATEST_TIME = TimeRange.WHOLE_DAY.end();
+
+  /* 
+    This finds all times where event could take place.
+    @param events Takes a collection of events
+    @param request Takes a MeetingRequest
+    @return Collection<TimeRange> This returns the times where the meeting can take place.
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Set<String> desiredGuests = new HashSet<>();
+    desiredGuests.addAll(request.getAttendees());
+    int duration = (int) request.getDuration();
+
+    if (duration > LATEST_TIME){
```

Comment:
Although it is working correctly in this case, usually we try to compare "duration" with "duration", not "duration" with "point in time". You can compare it with TimeRange.WHOLE_DAY.duration